src/lib.rs

```diff
@@ -342,19 +351,21 @@ fn apply_filter_with_flags(bpf_filter: BpfProgramRef, flags: libc::c_ulong) -> R
         return Err(Error::EmptyFilter);
     }
 
+    let bpf_prog = sock_fprog {
```

Comment:
The kernel will not allow installation of filters larger than 4096 instructions: https://man7.org/linux/man-pages/man2/seccomp.2.html.

We are checking for this limit when compiling the BPF. Search for: `BPF_MAX_LEN`.

But indeed, we shouldn't perform the lossy cast with `as`. We can keep the check you added, just make it use the 4096 limit

Reply:
Done

Reply:
please update also the PR description and squash commits into one with a clear message